### PR TITLE
Add cowsay-operator to the repository

### PR DESCRIPTION
**Pull request by zen-dog**
used as a demonstration and test for KUDO pipe-tasks.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Binaries for programs and plugins
+.idea
```
Comment (porridge) [inline .gitignore]:
```suggestion
/.idea
```

File: repository/cowsay/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# KUDO Cowsay operator
+
+The KUDO cowsay operator is a small demo for the KUDO [Pipe-Tasks](https://github.com/kudobuilder/kudo/blob/master/keps/0017-pipe-tasks.md).
+
+### Overview
+
+KUDO Cowsay operator:
+
+- Uses KUOD pipe-tasks and [cowsay.morecode.org](http://cowsay.morecode.org) to generate a customized index.html
```
Comment (porridge) [inline repository/cowsay/README.md]:
```suggestion
- Uses KUDO pipe-tasks and [cowsay.morecode.org](http://cowsay.morecode.org) to generate a customized `index.html`
```

File: repository/cowsay/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - name: genwww
+    kind: Pipe
+    spec:
+      pod: pipe-pod.yaml
+      pipe:
+        - file: /tmp/indexx.html
+          kind: ConfigMap
+          key: indexHtml
+```
+- Launched an nginx webserver with it
```
Comment (porridge) [inline repository/cowsay/README.md]:
```suggestion
- Launches an nginx webserver with it
```

File: repository/cowsay/operator/operator.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+apiVersion: kudo.dev/v1beta1
+name: "cowsay"
+version: "0.1.0"
+kubernetesVersion: 1.13.0
+maintainers:
+  - name: zen-dog
+    email: <adukhovniy@d2iq.com>
```
Comment (porridge) [inline repository/cowsay/operator/operator.yaml]:
```suggestion
    email: <adukhovniy@D2iQ.com>
```
I think that's the spelling marketing wants us to use.

File: repository/cowsay/operator/templates/pipe-pod.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Pod
+spec:
+  volumes:
+  - name: shared-data
+    emptyDir: {}
+
```
Comment (porridge) [inline repository/cowsay/operator/templates/pipe-pod.yaml]: Remove blank line?

**Review by porridge — APPROVED** [review]

File: repository/cowsay/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+KUDO Cowsay operator:
+
+- Uses KUOD pipe-tasks and [cowsay.morecode.org](http://cowsay.morecode.org) to generate a customized index.html 
+```yaml
+  - name: genwww
+    kind: Pipe
+    spec:
+      pod: pipe-pod.yaml
+      pipe:
+        - file: /tmp/indexx.html
```
Comment (alenkacz) [inline repository/cowsay/README.md]: isn't it rather `index.html`?

File: repository/cowsay/operator/operator.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    kind: Apply
+    spec:
+      resources:
+        - deployment.yaml
+  - name: genwww
+    kind: Pipe
+    spec:
+      pod: pipe-pod.yaml
+      pipe:
+        - file: /tmp/indexx.html
```
Comment (alenkacz) [inline repository/cowsay/operator/operator.yaml]: index

**Review by alenkacz — APPROVED** [review]
LGTM minus the typo. I think every operator should automatically come with a test (at least some basic one) but this is not possible here right now before we release 0.9.0 so I am letting this one go :-)

And also - MOOO 🐮

File: repository/cowsay/operator/templates/pipe-pod.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  initContainers:
+    - name: init
+      image: busybox
+      command: [ "/bin/sh", "-c" ]
+      args:
+        - wget -O /tmp/index.html 'http://cowsay.morecode.org/say?message=Good+things+come+when+you+least+expect+them&format=html'
+      volumeMounts:
+        - name: shared-data
+          mountPath: /tmp
```
Comment (nfnt) [inline repository/cowsay/operator/templates/pipe-pod.yaml]: Nit: add a return

**Review by nfnt — APPROVED** [review]
```
  ________________________________
/ Moo! Only a nit and the existing \
\ comments.                        /
  --------------------------------
         \   ^__^ 
          \  (oo)\_______
             (__)\       )\/\
                 ||----w |
                 ||     ||
```

File: repository/cowsay/README.md (hunk 6 above)
Comment (zen-dog) [reply repository/cowsay/README.md]: Yes, it is! God bless reviews! 🙏 😂 (this was from a test of what happens when the referenced file is missing)

**Comment by zen-dog** [comment]
> I think every operator should automatically come with a test (at least some basic one) but this is not possible here right now before we release 0.9.0 so I am letting this one go :-)

Exactly! I'll add a test once we release 0.9.0!

**Pull request merged**